### ISSUE TITLE
Small usability tweak - keep selected table name when changing to Database Structure tab

### ISSUE THREAD

**Issue by justinclift** [sqlitebrowser/sqlitebrowser#1077]
### Details for the issue

Just noticed a small usability awkwardness when creating a bunch of sample databases.

If I'm on the Browse Data tab and have a table selected, when I switch to the Database Structure tab the selection is lost.  So, I have to reselect the table name in the Database Structure tab before being able to click the Modify table button.

This is a bit of a pain when going back and forth between the two tabs, modifying the database table each iteration for whatever reason. eg changing data types to match imported data

#### I'm opening this issue because:

- [x] DB4S needs a feature

#### I'm using DB4S on:

- [x] Linux: ( Fedora 25 x64 )

#### I'm using DB4S version:

- [x] Latest git master 7195acf4885d57cbd1a3c17d083e6140d87538f6

**Comment by chrisjlocke** [comment]
Have you downloaded the latest nightly?  We've recently..... oh, its you ... force of habit..! ;)

It doesn't do this on Windows version.  Well, it does, but using different steps.

Database Structure tab. Select table.  Change tab to Browse Data.  Click back to Database Structure.  Selection is maintained.  Click Modify Table.  Click Cancel.  Selection is lost.  **If** OK is selected, however, then the selection is maintained.

**Comment by justinclift** [comment]
Ahhh, interesting.  For me on Linux the selection is lost even when the OK but is clicked.  Modifying field types, from the default TEXT to INTEGER in this case.

**Comment by chrisjlocke** [comment]
You're right.  Clicking OK with no changes does nothing.  As soon as I add a new field or make an amendment, **then** the selection disappears, before I've even clicked OK or Cancel.

**Comment by justinclift** [comment]
Yeah.  Having done this quite a few times over the last few hours now :wink:, I think I'm getting an idea of what the cause might be.

When I change a column field type from it's current setting to something new (eg TEXT -> INTEGER), the cursor/focus goes into the new cell text area, and the table selection disappears at the same time.

I think the cursor/focus change thing is so people can type in their own custom types, which is required for some scenarios.

The way to fix this is probably by us somehow keeping track of the selected table when entering the modify table dialog, and then reselecting that when the dialog closes.  That's a wild arse guess though. :smile:
